yakultbottle iP

yakultbottle opened the pull request.

yakultbottle committed:
Rename Duke to Jeremy

yakultbottle committed:
Add basic greet and exit

yakultbottle committed:
Add ASCII art of Jeremy

yakultbottle committed:
Abstract print statements into their own class

yakultbottle committed:
Implement command echoing and exit functionality

- Echo user commands back to the console
- Exit the program when the user types 'bye'

yakultbottle committed:
Add feature to store user-entered text and display it on request

yakultbottle committed:
Abstract list functionality into own class

yakultbottle committed:
Refactor Print class as PrintUtils class

Add print and println methods to PrintUtils class such that all prints
get called from the PrintUtils now. This is more clear than previously,
as custom prints would still call System.out. This makes the code neater
and cleaner.

Rename variables and methods to follow the coding standard

yakultbottle committed:
Add functionality to mark and unmark tasks as done

- Abstract functionality of Task into Task class
- Update List class to use an array of Task instead of Strings
- Refactor main loop to use switch case for command branching
- Add string parsing to handle mark and unmark commands with numbers

yakultbottle committed:
Correct boundary check, abstract logic, clean up

- Corrected faulty boundary check in List.java that was "taskNumber >=
  tasks.length", which is always 100, and changed it to "taskNumber >=
  size"
- Abstracted logic of maskTaskAsDone and markTaskAsNotDone into a
  separate method that they both call with a boolean
- Abstracted logic of printing status icon into getStatusIcon
- Changed name of taskDescription variable into description

yakultbottle committed:
Update Jeremy's responses for Tasks

- Change the message for invalid Task number
- Change the message for adding and unmarking tasks
- Separate linebreak from the printTask method for clarity

yakultbottle committed:
Refactor Task class with toString method

Previously Task class had a method printTask(), which itself called
PrintUtils. This led to some ambiguity, and removed flexibility in how
it was being handled, ie if you wanted to further format it before
printing. Returning a String with toString makes handling printing more
outright and straightforward

Also modified setTaskStatus to reduce nesting

yakultbottle committed:
Add Todo, Event, Deadline classes extending Task

- Created three new classes (Todo, Event, Deadline) that extend the Task class
- Moved string processing for task arguments into the constructors of these classes. This approach keeps control flow in Jeremy.java cleaner, despite making constructors more complex
- Retained the default behavior of creating a generic Task for unrecognized commands. Note: This default behavior is not intended to be permanent and may be revised in the future

yakultbottle committed:
Add automated UI tests

yakultbottle committed:
Rename List class to TaskList, cleanup

Cleanup
- Change some variables to be final
- Change the formatting of the logo string

yakultbottle committed:
Refactor switch statement with Command enum

Change the switch statement from a String to a enum I defined as
Command. This inherently removes the "default" case, as any error
handling is handled by the enum. As such, it now handles errors using
the UNKNOWN_COMMAND value.

yeekian commented on src/main/java/Jeremy.java:
```diff
@@ -0,0 +1,48 @@
+            String argument = parts.length > 1 ? parts[1] : "";
+            Command command = Command.fromString(commandStr);
+
+            switch (command) {
```
I like that your switch case follows the coding standard and the "case" is flushed with "switch".

yeekian commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,19 @@
+
+        String[] parts = input.split("/", 3);
+
+        this.from = parts[1].trim().substring(5); // ignore "from "
```
May I ask why you chose to use inline comments instead of block comments? Nontheless, I like that your comments are short :)

yakultbottle replied:
I use inline comments whenever my comments only apply to a single line of code, as it is sometimes ambiguous as to how many lines of code a block comment is referring to. I am unsure if this is best practice though

yeekian commented on src/main/java/Jeremy.java:
```diff
@@ -0,0 +1,48 @@
+                taskList.markTaskAsDone(Integer.parseInt(argument));
+                break;
+            case UNMARK:
+                taskList.markTaskAsNotDone(Integer.parseInt(argument));
```
Should line 25 be indented to match line 24?

yakultbottle replied:
I believe the indentation issue might just be hard to see in the review tab due to the length of this line of code. From what I can tell, the indentation looks correct. Please double-check in the Conversations tab in case I misunderstood your comment

yeekian commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,26 @@
+    }
+
+    public String toString() {
+        return "[" + getStatusIcon() + "] " + description;
```
I like that you used polymorphism to print out the task description 👍

yeekian commented on src/main/java/Jeremy.java:
```diff
@@ -0,0 +1,48 @@
+                taskList.markTaskAsDone(Integer.parseInt(argument));
+                break;
+            case UNMARK:
+                taskList.markTaskAsNotDone(Integer.parseInt(argument));
```
Should this be blank line will be removed?

yakultbottle replied:
There is no blank line in this section of code, I think again this is due to the text display on GitHub being a little weird at this part

yeekian reviewed (commented):
Overall good job with following the coding standard. Keep up the good work!

Zhang-Zhitong commented on src/main/java/PrintUtils.java:
```diff
@@ -0,0 +1,64 @@
+
+    private static final String NAME = "Jeremy";
+
+    private static final String LOGO = """
```
Something underhere seems sus yakultbottle btw, anyone knows anything?

Zhang-Zhitong replied:
check first letter of each word

Zhang-Zhitong commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,56 @@
+public class TaskList {
```
I like the good use of nouns for class names

Zhang-Zhitong commented on src/main/java/TaskList.java:
```diff
@@ -0,0 +1,56 @@
+        PrintUtils.lineBreak();
+    }
+
+    public void markTaskAsDone(int taskNumber)  {
```
I like the good use of encapsulation

Zhang-Zhitong commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,26 @@
+public class Task {
```
Could this be made into an abstract class?

yakultbottle replied:
This is a good catch, thanks for this. I'll probably refactor this next week

Zhang-Zhitong reviewed (commented):
Looks good to me. cant find anything to nit.

yakultbottle committed:
Update automated tests

yakultbottle committed:
Add error handling to mark and unmark

- Add handling of mark and unmark commands with invalid numbers with the
use of an exception, handled in the setTaskStatus method
- Refactor prior error handling for checking of valid taskNumber to use
exceptions

yakultbottle committed:
Create custom exceptions to handle errors

- Create JeremyException class for other custom exceptions to inherit
from
- Create exceptions for empty argument, illegal command, and illegal
command format
- Refactor the switch-case to use try-catch to handle exceptions there

yakultbottle committed:
Refactor setTaskStatus to use custom exceptions

yakultbottle committed:
Merge branch 'branch-Level-5'

yakultbottle committed:
Organise classes into packages

yakultbottle committed:
Merge branch 'branch-A-Packages'

yakultbottle committed:
Refactor constants in PrintUtil class

- Change LINE_BREAK constant to use .repeat method from String
- Change NAME constant from jeremy.Jeremy to Jeremy, which is a
  byproduct of lazy refactoring from IntelliJ

yakultbottle committed:
Fix bad refactor

yakultbottle committed:
Change TaskList to use ArrayList instead of Array

yakultbottle committed:
Add delete function to TaskList

yakultbottle committed:
Add basic save to file

yakultbottle committed:
Reformat saved strings to be more readable

yakultbottle committed:
Add read data functionality

yakultbottle committed:
Add handling for data folder not existing yet

yakultbottle committed:
Merge branch 'branch-Level-6'

yakultbottle committed:
Merge branch 'branch-Level-7'

yakultbottle committed:
Merge branch 'branch-Level-7'

yakultbottle committed:
Merge conflict resolution

Didn't check before pushing to master, so there's an extra commit to
resolve the merge conflict

yakultbottle committed:
Fix file storage creation

thomasjlalba commented on src/main/java/jeremy/Jeremy.java:
```diff
@@ -0,0 +1,89 @@
+                    break;
+                case EVENT:
+                    taskList.addTask(new Event(argument));
+                    break;
```
Understand you have a catch for commands not found before it enters the switch statement, but maybe you could still [include a default branch for the switch statements](https://nus-cs2113-ay2425s1.github.io/website/se-book-adapted/chapters/codeQuality.html#use-the-default-branch)

thomasjlalba commented on src/main/java/jeremy/util/PrintUtils.java:
```diff
@@ -0,0 +1,65 @@
+package jeremy.util;
+
+public class PrintUtils {
```
Excellent job on storing these strings and not using them as Magic strings throughout the code!

thomasjlalba commented on src/main/java/jeremy/Storage.java:
```diff
@@ -0,0 +1,88 @@
+            }
+
+            switch (command) {
+            case "T":
```
Maybe you could avoid magic strings and indicate what it is?

thomasjlalba commented on src/main/java/jeremy/Jeremy.java:
```diff
@@ -0,0 +1,89 @@
+package jeremy;
+
+import jeremy.command.Command;
+import jeremy.exception.*;
```
Maybe you can avoid * and list explicitly the import class as mentioned [here](https://se-education.org/guides/conventions/java/basic.html#statements)

thomasjlalba reviewed (commented):
Great job! Just some minor nitpicks

yakultbottle committed:
Refactor main file to be more OOP

- Move what was everything in main method for Jeremy.java to run() and
  constructor
- Renamed PrintUtils to Ui
- Instantiate a Storage object
- Change many methods to not be static

yakultbottle committed:
Change Task to be an abstract class

yakultbottle committed:
Clean up error handling to have less clutter in main

yakultbottle committed:
Address feedback from Thomas

- Add default branch to switch statements
- Avoid magic strings by converting into Command enum
- Change all wildcard imports

yakultbottle committed:
Remove verbose printing when loading tasks

yakultbottle committed:
Abstract Parser class

yakultbottle committed:
Refactor Command to be abstract class with inherited subclasses

yakultbottle committed:
Add date parsing

yakultbottle committed:
Update UI tests to use new date system

yakultbottle committed:
Add find command

yakultbottle committed:
Add Javadoc comments

yakultbottle committed:
Merge pull request #1 from yakultbottle/branch-Level-8

Branch level 8

yakultbottle committed:
Merge branch 'master' into branch-Level-9

yakultbottle committed:
Merge pull request #2 from yakultbottle/branch-Level-9

Add find command

yakultbottle committed:
Merge branch 'master' into branch-A-Javadoc

yakultbottle committed:
Merge pull request #4 from yakultbottle/branch-A-Javadoc

Branch A-Javadoc

yakultbottle committed:
Change find output to return the index of each found item

yakultbottle committed:
Update README

yakultbottle committed:
Fix broken formatting in README

yakultbottle committed:
Update README.md to fix formatting

yakultbottle committed:
Fix bug for printing messages when loading tasks

yakultbottle committed:
Merge pull request #5 from yakultbottle/branch-cleanup

Fix bug for printing messages when loading tasks

yakultbottle committed:
Fix some bad Exception printing

yakultbottle committed:
Fix inconsistency between Event and Deadline
